pandas/tests/test_generic.py

```diff
@@ -1539,6 +1540,15 @@ def test_to_xarray(self):
                            expected,
                            check_index_type=False)
 
+    def test_deepcopy_empty(self):
```

Comment:
@bmcfee you can add the test you wanted. but you are making fundamental changes. These HAVE to be tested on all classes that are affected. (and you ARE affecting everything)